vulkan_helpers/vulkan_application.h

```diff
@@ -406,8 +413,21 @@ class VulkanApplication {
     // writes are visible to the GPU
     void flush(size_t offset, size_t size) {
       if (flush_memory_range_) {
+        auto adjusted_offset = offset_ + offset;
+        auto delta = adjusted_offset % kMaxNonCoherentAtomSize;
+        // Make offset a multiple of kMaxNonCoherentAtomSize (rounding down)
+        adjusted_offset -= delta;
+        // Adjust the size up to compensate
+        auto adjusted_size = size + delta;
+        // And now make the size a multiple of kMaxNonCoherentAtomSizes
```

Comment:
I think we also have to make sure that upper bound <= the size of the memory range if we do this. I.e. I dont think we can (legally) flush anything outside outside of the mapped memory range.

Reply:
Aha, I forgot that we extended the size of the buffer originally. Thanks!